reactor-core/src/main/java/reactor/core/publisher/SerializedSubscriber.java

```diff
@@ -69,21 +69,23 @@ public void onSubscribe(Subscription s) {
 	@Override
 	public void onNext(T t) {
 		if (cancelled || done) {
+			Operators.onDiscard(t, actual.currentContext());
```

Comment:
Should not we use `Operators.onNextDropped` here?

operators `discard` is `Reactor` only feature, so someone who uses another framework (yeah, if reactor meets non-reactor publisher, then `SerializedSubscriber` applies) then it is going to be impossible to handle discarded element at all.

Also, I found it really confusing having onDrop and `discard` unintegrated, which means that if no on discard hooks, most likely onDrop should be applied and element should find at least something

Reply:
right ! one way to do it would be to check for `cancelled` and `done` separately.